src/cdk/coercion/string-array.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * Coerces a value to an array of trimmed non-empty strings.
+ * Anything that is not an array or null-ish will be turned into a string via a template literal.
+ * Null-ish values will result in an empty array.
+ * This results in the following outcomes:
+ * - `null` -&gt; `[]`
+ * - `[null]` -&gt; `["null"]`
+ * - `[1, [2, 3]]` -&gt; `["1", "2,3"]`
+ * - `[{ a: 0 }]` -&gt; `["[object Object]"]`
+ * - `{ a: 0 }` -&gt; `["[object", "Object]"]`
+ *
+ * Useful for defining CSS classes or table columns.
+ * @param value the value to coerce into an array of strings
+ * @param separator split-separator if value isn't an array
+ */
+export function coerceStringArray(value: any, separator: string | RegExp = /\s+/): string[] {
+  const result = [];
+
+  if (value != null) {
+    const sourceValues = Array.isArray(value) ? value : `${value}`.split(separator);
+    for (const sourceValue of sourceValues) {
+      const trimmedString = `${sourceValue}`.trim();
+      if (!!trimmedString) {
```

Comment:
The `!!` is necessary when using the expression in a conditional since the value isn't used anywhere; you only need `!!` if you're passing the value along to something that expects a `boolean`

Reply:
I'm used to use that to clarify that the variable is not actually a boolean. But you are right, in this case that's not really necessary as you can see where the value comes from.